### PR TITLE
Add GAPIC Bazel Extensions for PHP

### PR DESCRIPTION
**Pull request by vam-google**
**Usage instructions:**

In `WORKSPACE` file:
```python
##############################################################################
# PHP
##############################################################################

load("@com_google_api_codegen//rules_gapic/php:php_gapic_repositories.bzl", "php", "php_gapic_repositories")

php(
    name = "php",
    version = "7.1.30",
)

php_gapic_repositories()
```

In `BUILD.bazel` file (using `googleapis/cloud/language/v1/BUILD.bazel` as example:
```python
##############################################################################
# PHP
##############################################################################
load(
    "@com_google_googleapis_imports//:imports.bzl",
    "php_proto_library",
    "php_grpc_library",
    "php_gapic_library",
    "php_gapic_assembly_pkg",
)

php_proto_library(
    name = "language_php_proto",
    deps = [":language_proto"],
)

php_grpc_library(
    name = "language_php_grpc",
    srcs = [":language_proto"],
    deps = [":language_php_proto"],
)

php_gapic_library(
    name = "language_php_gapic",
    src = ":language_proto_with_info",
    gapic_yaml = "language_gapic.yaml",
    service_yaml = "//google/cloud/language:language_v1.yaml",
    deps = [
        ":langauge_php_grpc",
        ":language_php_proto",
    ],
)

# Opensource Packages
php_gapic_assembly_pkg(
    name = "google-cloud-language-v1-php",
    deps = [
        ":language_php_gapic",
        ":language_php_grpc",
        ":language_php_proto",
    ],
)
```

### CODE REVIEW

File: rules_gapic/php/php_gapic_repositories.bzl [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Copyright 2018 Google LLC
```
Comment (michaelbausor) [inline rules_gapic/php/php_gapic_repositories.bzl]: 2019

File: rules_gapic/README.md [hunk 2]
```diff
@@ -19,30 +20,44 @@ The rules will call `gapic-generator` and do all the necessary pre- and post- ge
...
 
 #### Go
-1. **`go/go_gapic_repositories.bzl`** - this file declares the Go-specific dependencies of the generated output and is supposed to be included in the WORKSPACE of the consuming workspace (for examples in `googleapis`).
\ No newline at end of file
+1. **`go/go_gapic_repositories.bzl`** - this file declares the Go-specific dependencies of the generated output and is supposed to be included in the WORKSPACE file of the consuming workspace (for examples in `googleapis`).
+
+#### PHP
+1. **`go/php_gapic_repositories.bzl`** - this file declares the PHP-specific dependencies of the generated output and is supposed to be included in the WORKSPACE file of the consuming workspace (for examples in `googleapis`). This file also declares the `php` repository rule, which downloads and builds from sources PHP interpreter (by using `gcc`, `make` and `autoconf` tools, so they are expected to be installed on the system).
```
Comment (eoogbe) [inline rules_gapic/README.md]: php/php_gapic_repositories.bzl

**Comment by vam-google** [comment]
@michaelbausor Probably the easiest way would be to wait till it is pushed, then point to the new version of the gapic generator in your `WORKSPACE` file.

Also you would need to add the following PHP-specific dependencies in your `WORKSPACE` file:
```python
##############################################################################
# PHP
##############################################################################

load("@com_google_api_codegen//rules_gapic/php:php_gapic_repositories.bzl", "php", "php_gapic_repositories")

php(
    name = "php",
    version = "7.1.30",
)

php_gapic_repositories()
```

This will download and build php interpreter on your machine. I'm still wondering how much of C++ dependencies that would need on a vanilla machine, but it should be a bunch of `apt-get installs` of thypical `-dev` libs (if they are not already installed) and installation of `autoconf`. Bazel has plans on Q3 to make C++ builds fully hermetic, so hopefully this can be mitigated in future.

**Comment by vam-google** [comment]
@michaelbausor Please see updated PR description for more details on how to use this. Also please note that it implies that the  new rules were added in the `imports` repository rule in googlapis (is trivial, I'll submit those right after this PR is merged).

**Comment by michaelbausor** [comment]
@vam-google SGTM

**Pull request merged**